source/MaterialXRender/Handlers/StbImageLoader.cpp

```diff
@@ -73,12 +78,16 @@ bool StbImageLoader::saveImage(const FilePath& filePath,
     {
         returnValue = stbi_write_hdr(filePathName.c_str(), w, h, channels, static_cast<float*>(data));
     }
+
+    if (yFlip)
+    {
+        stbi__flip_vertically_on_write = prevFlip;
+    }
     return (returnValue == 1);
 }
 
-bool StbImageLoader::acquireImage(const FilePath& filePath,
-                                  ImageDesc& imageDesc,
-                                  bool /*generateMipMaps*/)
+bool StbImageLoader::acquireImage(const FilePath& filePath, ImageDesc &imageDesc,
+                                  const HwImageDescRestrictions* /*restrictions*/) 
```

Comment:
Restrictions on base type or channel count is not considered here. If this use of a restrictions specifier is to be used I guess all image loaders should respect it?